src/Search/Processors/SearchUpdateProcessor.php

```diff
@@ -84,13 +94,35 @@ protected function prepareIndexes()
 
                 // Ensure that indexes for all new / updated objects are included
                 $objs = DataObject::get($base)->byIDs(array_keys($ids));
+                $objectPermissions = [];
+
+                // Attempt to use optimised permission checker if present.
+                // NOTE: This reduces the scope of the check to DB-based permissions (custom canView logic is ignored)
```

Comment:
Since this is the CMS->Save() code path, we should to just use the standard and more accurate canView() check instead of the getPermissionChecker here since performance is much less of a concern

Reply:
I think it makes more sense to keep the behaviour consistent across the two - there are already enough edgecases as it is. Any thoughts on this @chillu?